tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,11 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return self.__repr_()
+
+  def __repr__(self):
+    return "<tf.Variable '%s' shape=%s dtype=%s>" % (
+            self.name(), self.get_shape(), self.dtype.name)
```

Comment:
Thanks for making that change. I just noticed that it should be `self.name` instead of `self.name()` (for the same reasons as `self.dtype`...). Can you please make that change as well?

This little piece of code has proven surprisingly tricky for such a small change! I think it might be a good idea to add a unit test, which would have caught these issues. Can you please add a test of `str(tf.Variable(...))` and `repr(tf.Variable(...))` to `variables_test.py`? You can base it on the similar tests for `tf.Operation` [here](https://github.com/tensorflow/tensorflow/blob/286a5361df528dd5d5f1c5b319a69a0029e245bc/tensorflow/python/framework/ops_test.py#L344).